testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1264,28 +1264,24 @@ public void testFederatedMaterializedView()
 
                     assertThat(query("TABLE " + viewName))
                             // The MV is not refreshed yet, so gets inlined
-                            .hasPlan(readFromBaseTables)
-                            .matches("VALUES (VARCHAR 'first_table', BIGINT '1')");
+                            .hasPlan(readFromBaseTables);
 
                     assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
                     assertThat(query("TABLE " + viewName))
-                            .hasPlan(readFromStorageTable)
-                            .matches("VALUES (VARCHAR 'first_table', BIGINT '1')");
+                            .hasPlan(readFromStorageTable);
 
                     // Change underlying state
                     mockListing.set(List.of("first_table", "second_table"));
 
                     // The materialized view should return now-stale data, since it doesn't know when it is no longer fresh
                     assertThat(query("TABLE " + viewName))
-                            .hasPlan(readFromStorageTable)
-                            .matches("VALUES (VARCHAR 'first_table', BIGINT '1')");
+                            .hasPlan(readFromStorageTable);
                     assertThat(query("SELECT freshness FROM system.metadata.materialized_views WHERE schema_name = CURRENT_SCHEMA AND name = '" + viewName + "'"))
                             .matches("VALUES VARCHAR 'UNKNOWN'");
 
                     assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 2);
                     assertThat(query("TABLE " + viewName))
-                            .hasPlan(readFromStorageTable)
-                            .matches("VALUES (VARCHAR 'first_table', BIGINT '1'), ('second_table', 1)");
+                            .hasPlan(readFromStorageTable);
```

Comment:
I don't think removing `matches` is safe here. `matches` is redundant only if we want to verify result correctness with and without pushdown. In this case the result doesn't depend on pushdown so explicit `matches` is useful.

Consider a case where the query starts returning different results - since the verification that `hasPlan` does is also performed by executing the query on Trino itself the results will always match.

Reply:
Exactly, you are right.
Removed modification for the case that `matches` is used with `hasPlan`